components/org.wso2.carbon.consent.mgt.ui/src/main/resources/web/consent/add-purpose.jsp

```diff
@@ -79,6 +84,23 @@
         if (localClaims != null) {
             claims.addAll(Arrays.asList(localClaims));
         }
+
+        String currentUser = (String) session.getAttribute("logged-user");
+        ConsentManagementServiceClient serviceClient = new ConsentManagementServiceClient(currentUser);
+        List<String> purposeIDs;
+        purposeIDs = Arrays.asList(StringUtils.split(StringUtils.substringBetween(purposeIdList, "[", "]"), ", "));
```

Comment:
use a parser to parse the json response.

Reply:
changed to use JSONArray to get the string parsed to an JSONarray